include/zephyr/drivers/emul.h

```diff
@@ -89,7 +90,16 @@ struct emul {
 extern const struct emul __emul_list_start[];
 extern const struct emul __emul_list_end[];
 
-/* Use the devicetree node identifier as a unique name. */
+/**
+ * @brief Use the devicetree node identifier as a unique name.
+ */
+#define EMUL_DT_NAME_GET(node_id) (_CONCAT(__emulreg_, node_id))
+
+/**
+ * @brief Use the devicetree node identifier as a unique name.
```

Comment:
not sure there's any need to deprecate this as the emulator API isn't stable.  So we can just remove `EMUL_REG_NAME`.